lib/discordgo/tools/cmd/permnames/main.go

```diff
@@ -0,0 +1,129 @@
+package main
```

Comment:
This code is admittedly complicated but the current discordgo code already relies on some codegen (`eventhandlers`) and we need not touch the code ever again after it is committed.